### PR TITLE
New bows/crossbows

### PR DESCRIPTION
**Pull request by darktoes**
#### Summary

```SUMMARY: Content "Adds new bows and crossbows"```

#### Purpose of change
Adds more content, specifically bows and crossbows for players that like those type of weapons.
Greatbows have been on my personal todo-list for a while, as a strength-based ranged weapon solution for characters with exceptionally high (minimum 18) strength.

There has been a number of discussions on the fact that crossbows ingame are lacklustre, so I added two new ones which are nearly straight upgrades to the base crossbow. Also removed a few mod slots from the base crossbow for realism, since it seems to be intended as a primitive medieval design.

Gave the huge crossbow slightly higher damage since it's worthless otherwise. Probably still is.

Fixed dampening kits on crossbows (oops).

Added the hybrid longbow because they're quite common in real life and make a good middle-ground for players between the longbow and recurve.

#### Describe the solution
Greatbows: They're around 3 times as powerful, as discussed with Kevin, although that will vary as they use standard arrows as ammunition, which may be changed in future depending on feedback, and if a projectile penetration mechanic is ever introduced.

Crossbows: Composite and Compound variants. Composite is stronger with more mod slots, but takes longer to reload. Compound is stronger again but actually takes less time to reload than the original crossbow. Intended to make crossbows a viable endgame(ish) weapon, and eventually act as a lucky find for crossbow oriented characters in firearm stores.

#### Additional context
I intend to eventually revamp itemgroups for all the archery equipment, removing the couple of archery item groups and adding archery equipment to sportsy places and museums, as well as the compound bow and compound crossbow to firearm stores. That will have to be another PR though.

### CODE REVIEW

File: data/json/items/archery.json [hunk 1]
```diff
@@ -540,7 +542,40 @@
...
+      "reload": 40,
+      "valid_mod_locations": [["dampening", 1]]
+   },
+   {
+      "id" : "hybridbow",
+      "type" : "GUN",
+      "symbol" : "(",
+      "color" : "yellow",
+      "name" : "hybrid longbow",
+      "description" : "A modernised, six-foot longbow that curves away from the archer at the tips for increased power. It can be used effectively by those of somewhat above-average strength. Designed for archers that appreciate the smoothness of a traditional longbow, but desire the superior speed of a modern recurve.",
```
Comment (Night-Pryanik) [inline data/json/items/archery.json]: Two spaces after periods between the sentences.

Comment (darktoes) [reply data/json/items/archery.json]: Damn. Thought i got all those.

File: data/json/recipes/recipe_weapon.json [hunk 2]
```diff
@@ -403,6 +419,35 @@
...
+      { "id": "HAMMER_FINE", "level": 1 },
+      { "id": "SAW_M_FINE", "level": 1 },
+      { "id": "CHISEL", "level": 3 }
+    ],
+    "tools": [ [ [ "swage", -1 ] ], [ [ "tongs", -1 ] ] ],
+    "components": [
+      [ [ "stick", 10 ], [ "2x4", 6 ] ],
+      [ [ "steel_standard", 2, "LIST" ] ],
+      [ [ "plastic_chunk", 25 ] ],
+      [ [ "duct_tape", 600 ], [ "superglue", 24 ] ],
```
[the remark this answers is not in the document]
Comment (darktoes) [reply data/json/recipes/recipe_weapon.json]: Probably should be, but most of the other recipes in that file don't use it.

**Comment by darktoes** [comment]
Spider silk probably wouldn't make a bow string any better than standard materials. Spider mutants can apparently already make rope though.

**Comment by Mecares** [comment]
Well it can have better mechanical properties regarding toughness and elasticity than the materials currently used by in game characters.

**Comment by darktoes** [comment]
Except you don't want a bow string to be elastic, it should be as inelastic as possible, and durability isn't a very big concern. Modern materials work about as well as you need, using spider silk probably wouldn't make any difference.

File: data/json/items/crossbows.json [hunk 3]
```diff
@@ -173,7 +173,59 @@
...
+      "valid_mod_locations": [[ "accessories", 4 ],[ "grip", 1 ],[ "mechanism", 4 ],[ "rail", 1 ],[ "sights", 1 ],[ "stock", 1 ],[ "underbarrel", 1 ],[ "dampening", 1 ]]
+   },
+   {
+      "id": "compcrossbow",
+      "type": "GUN",
+      "symbol": "(",
+      "color": "green",
+      "name": "compound crossbow",
+      "reload_noise_volume": 4,
+      "description": "A high-power crossbow with shaped cams and extra cables for high-power shots.  Takes less time to reload due to a modern stringpulling system.",
```
[the remark this answers is not in the document]
Comment (darktoes) [reply data/json/items/crossbows.json]: Fixed.

File: data/json/items/archery.json [hunk 4]
```diff
@@ -571,7 +606,52 @@
...
+      "description" : "An extremely large and stiff longbow, made with enormous limbs and a thick string to take an enormous amount of energy.  Takes incredible strength to draw.",
+      "price" : 150000,
+      "min_strength" : 18,
+      "weight" : 3200,
+      "volume" : 25,
+      "armor_data" : {
+         "covers" : ["TORSO"],
+         "coverage" : 7,
+         "material_thickness" : 2,
+         "encumbrance" : 20
```
Comment (ZhilkinSerg) [inline data/json/items/archery.json]: Encumbrance value of 20 is not changed compare to "longbow".

File: data/json/items/archery.json [hunk 5]
```diff
@@ -571,7 +606,52 @@
...
+      "override" : true,
+      "type" : "GUN",
+      "name" : "compound greatbow",
+      "name_plural" : "compound greatbows",
+      "description" : "An extremely large and powerful compound bow, made with enormous limbs and a thick string to take an enormous amount of energy.  Takes incredible strength to draw.",
+      "price" : 160000,
+      "min_strength" : 18,
+      "weight" : 2000,
+      "volume" : 16,
+      "dispersion" : 100,
```
Comment (ZhilkinSerg) [inline data/json/items/archery.json]: Dispersion value of 100 is not changed compared to "compbow".

Comment (darktoes) [reply data/json/items/archery.json]: Removed that line.

File: data/json/items/archery.json (hunk 4 above)
Comment (darktoes) [reply data/json/items/archery.json]: Changed to 25, since it's bigger and heavier.

**Pull request merged**